File: Haskell/beginner/1828.hs
-- https://judge.beecrowd.com/pt/problems/view/1828
-- Bazinga!

import Control.Monad (replicateM_)

-- Main Function
main :: IO ()
main = do
    -- Get the number of tests
    testCasesStr <- getLine
    let testCases = read testCasesStr :: Int
    -- For every test, there is a scenario counter
    mapM_ (\i -> do
        scenarioStr <- getLine
        let (shellyStr, rajStr) = break (==' ') scenarioStr
        let shelly = shellyStr
        let raj = tail rajStr -- Remove the initial space from rajStr

        -- Takes the result and throw the winner at the console
        let gameResult = determineWinner shelly raj
        putStrLn $ "Caso #" ++ show i ++ ": " ++ gameResult
        ) [1..testCases] -- Creates a list from 1 to testCases


-- 
determineWinner :: String -> String -> String
determineWinner shelly raj
    | shelly == raj = "De novo!"
    | (shelly == "Spock" && raj == "pedra") = "Bazinga!"
    | (shelly == "lagarto" && raj == "Spock") = "Bazinga!"
    | (shelly == "tesoura" && raj == "lagarto") = "Bazinga!"
    | (shelly == "papel" && raj == "Spock") = "Bazinga!"
    | (shelly == "tesoura" && raj == "papel") = "Bazinga!"
    | (shelly == "papel" && raj == "pedra") = "Bazinga!"
    | (shelly == "lagarto" && raj == "papel") = "Bazinga!"
    | (shelly == "pedra" && raj == "lagarto") = "Bazinga!"
    | (shelly == "Spock" && raj == "tesoura") = "Bazinga!"
    | (shelly == "pedra" && raj == "tesoura") = "Bazinga!"
    | otherwise = "Raj trapaceou!"

-- By: Fernando Serra
-- https://github.com/fernandosserra

File: Python/beginner/1828.py
# https://judge.beecrowd.com/pt/problems/view/1828
# Bazinga!

# Test Cases
testCases = int(input())

for i in range(testCases):
    shelly, raj = input().split()
    
    if shelly == raj:
        print(f"Caso #{i + 1}: De novo!")
    elif shelly == "tesoura" and raj == "papel":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "papel" and raj == "pedra":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "pedra" and raj == "lagarto":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "lagarto" and raj == "Spock":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "Spock" and raj == "tesoura":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "tesoura" and raj == "lagarto":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "lagarto" and raj == "papel":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "papel" and raj == "Spock":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "Spock" and raj == "pedra":
        print(f"Caso #{i + 1}: Bazinga!")
    elif shelly == "pedra" and raj == "tesoura":
        print(f"Caso #{i + 1}: Bazinga!")
    else:
        print(f"Caso #{i + 1}: Raj trapaceou!")
        
# By: Fernando Serra
# https://github.com/fernandosserra